logpyle/__init__.py

```diff
@@ -821,11 +832,18 @@ def tick_after(self) -> None:
             self.save()
 
         # print watches
-        if self.tick_count >= self.next_watch_tick:
+        if self.tick_count+1 >= self.next_watch_tick:
             self._watch_tick()
 
         self.t_log += time() - tick_start_time
 
+        # Adjust log update time(s), t_log
+        for gd in self.after_gather_descriptors:
+            if isinstance(gd.quantity, LogUpdateDuration):
+                self._update_t_log(gd.quantity.name, gd.quantity())
```

Comment:
I'm not sure I prefer this to the previous solution (of lagging the value by one step).

- This doesn't measure the cost of writing out `LogUpdateDuration`, reducing accuracy.
- Treating it specially is ugly.

Reply:
> * This doesn't measure the cost of writing out `LogUpdateDuration`, reducing accuracy.

True, but this is (likely) just a single local update, so imho the accuracy isn't affected much. The accuracy isn't 100% in any case, since 
- the function call overhead of calling (e.g.) `tick_after` isn't measured
- most of the overhead of `self.t_log += time() - tick_start_time` isn't measured

> * Treating it specially is ugly.

Sure, but I think the code makes more sense in this way - gathering the logging time when logging is done instead of at the beginning of the next time step. Imho, having the values off by one step is really confusing, and can be discovered only by reading the code